File: C++/804. Unique Morse Code Words.cpp
class Solution {
public:
    int uniqueMorseRepresentations(vector<string>& words) {
        vector<string> tmp = {".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."};
        unordered_map<string, int> maple;
        for(auto word: words)
        {
            string s;
            for(auto c: word)
                s += tmp[c-'a'];
            maple[s] = 1;
        }
        return maple.size();
    }
};

File: Python/804. Unique Morse Code Words.py
class Solution:
    def uniqueMorseRepresentations(self, words):
        """
        :type words: List[str]
        :rtype: int
        """
        tmp = [".-","-...","-.-.","-..",".","..-.","--.","....","..",".---","-.-",".-..","--","-.","---",".--.","--.-",".-.","...","-","..-","...-",".--","-..-","-.--","--.."]
        ans = dict()
        for word in words:
            s = list()
            for c in word:
                s.append(tmp[ord(c) - ord('a')])
            ans[''.join(s)] = 1
        return len(ans)
            